src/transformers/models/lightglue/modeling_lightglue.py

```diff
@@ -0,0 +1,1149 @@
+    used to indicate which values in the keypoints, matches and matching_scores tensors are keypoint matching
+    information.
+
+    Args:
+        loss (`torch.FloatTensor` of shape `(1,)`, *optional*):
+            Loss computed during training.
+        matches (`torch.FloatTensor` of shape `(batch_size, 2, num_matches)`):
+            Index of keypoint matched in the other image.
+        matching_scores (`torch.FloatTensor` of shape `(batch_size, 2, num_matches)`):
+            Scores of predicted matches.
+        keypoints (`torch.FloatTensor` of shape `(batch_size, num_keypoints, 2)`):
+            Absolute (x, y) coordinates of predicted keypoints in a given image.
+        prune (`torch.IntTensor` of shape `(batch_size, num_keypoints)`):
+            Pruning mask indicating which keypoints are removed and at which layer.
+        mask (`torch.BoolTensor` of shape `(batch_size, num_keypoints)`):
+            Mask indicating which values in matches and matching_scores are keypoint matching information.
+        hidden_states (`Tuple[torch.FloatTensor, ...]`, *optional*):
+            Tuple of `torch.FloatTensor` (one for the output of each stage) of shape `(batch_size, 2, num_channels,
+            num_keypoints)` returned when `output_hidden_states=True` is passed or when
+            `config.output_hidden_states=True`
+        attentions (`Tuple[torch.FloatTensor, ...]`, *optional*):
+            Tuple of `torch.FloatTensor` (one for each layer) of shape `(batch_size, 2, num_heads, num_keypoints,
+            num_keypoints)` returned when `output_attentions=True` is passed or when
+            `config.output_attentions=True`
+    """
+
+    loss: Optional[torch.FloatTensor] = None
+    matches: Optional[torch.FloatTensor] = None
+    matching_scores: Optional[torch.FloatTensor] = None
+    keypoints: Optional[torch.FloatTensor] = None
+    prune: Optional[torch.IntTensor] = None
+    mask: Optional[torch.FloatTensor] = None
+    hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+    attentions: Optional[Tuple[torch.FloatTensor]] = None
+
+
+class LightGluePositionalEncoder(nn.Module):
+    def __init__(self, config: LightGlueConfig):
+        super().__init__()
+        self.projector = nn.Linear(2, config.descriptor_dim // config.num_attention_heads // 2, bias=False)
+
+    def forward(
+        self, keypoints: torch.Tensor, output_hidden_states: Optional[bool] = False
+    ) -> Union[Tuple[torch.Tensor], Tuple[torch.Tensor, torch.Tensor]]:
+        projected_keypoints = self.projector(keypoints)
+        cosines = torch.cos(projected_keypoints)
+        sines = torch.sin(projected_keypoints)
+        embeddings = torch.cat([sines, cosines], -1)
+        output = (embeddings, projected_keypoints) if output_hidden_states else (embeddings,)
+        return output
+
+
+# Copied from transformers.models.roformer.modeling_roformer.RoFormerSelfAttention with RoFormer->LightGlue
```

Comment:
I don't think we want to base it of Roformer, this depends on how involve it is for you but it's one of the "ugliest" code we have.